code/modules/cargo/console.dm

```diff
@@ -292,20 +275,35 @@
 
 /obj/machinery/computer/cargo/proc/generate_pack_data()
 	supply_pack_data = list()
-	for(var/pack in SSshuttle.supply_packs)
-		var/datum/supply_pack/P = SSshuttle.supply_packs[pack]
-		if(!supply_pack_data[P.group])
-			supply_pack_data[P.group] = list(
-				"name" = P.group,
+
+	if(!current_ship.docked_to)
+		return supply_pack_data
+
+	var/datum/overmap/outpost/outpost_docked = current_ship.docked_to
+
+	if(!istype(outpost_docked))
+		return supply_pack_data
+
+	for(var/datum/supply_pack/current_pack as anything in outpost_docked.supply_packs)
+		if(!supply_pack_data[current_pack.group])
+			supply_pack_data[current_pack.group] = list(
+				"name" = current_pack.group,
 				"packs" = list()
 			)
-		if((P.hidden))
+		if((current_pack.hidden))
+			continue
+		var/same_faction = current_pack.faction ? current_pack.faction.allowed_faction(current_ship.faction_datum) : FALSE
+		var/discountedcost = (same_faction && current_pack.faction_discount) ? current_pack.cost - (current_pack.cost * (current_pack.faction_discount * 0.01)) : null
+		if(current_pack.faction_locked && !same_faction)
 			continue
-		supply_pack_data[P.group]["packs"] += list(list(
-			"name" = P.name,
-			"cost" = P.cost,
-			"id" = pack,
-			"desc" = P.desc || P.name // If there is a description, use it. Otherwise use the pack's name.
+		supply_pack_data[current_pack.group]["packs"] += list(list(
+			"name" = current_pack.name,
+			"cost" = current_pack.cost,
+			"discountedcost" = discountedcost ? discountedcost : null,
+			"discountpercent" = current_pack.faction_discount,
+			"faction_locked" = current_pack.faction_locked, //this will only show if you are same faction, so no issue
+			"ref" = REF(current_pack),
+			"desc" = (current_pack.desc || current_pack.name) + (discountedcost ? "\n-[current_pack.faction_discount]% off due to your faction.\nWas [current_pack.cost]" : "") + (current_pack.faction_locked ? "\nYou are able to purchase this item due to your faction." : "") // If there is a description, use it. Otherwise use the pack's name.
```

Comment:
```suggestion
			"desc" = (current_pack.desc || current_pack.name) + (discountedcost ? "\n-[current_pack.faction_discount]% off due to your faction affiliation.\nWas [current_pack.cost]" : "") + (current_pack.faction_locked ? "\nYou are able to purchase this item due to your faction affiliation." : "") // If there is a description, use it. Otherwise use the pack's name.
```